### ISSUE TITLE
Support separate process and total regexes

### ISSUE THREAD

**Issue by Modelmat** [chriskiehl/Gooey#547]
I use `gs` to do some processing. Unfortunately, as far as I can tell, the current progress bar requires one progress_regex for both the total and the current.
`gs`'s output looks like:
```
Processing pages 1 through 1312.
Page 1
Page 2
Page 3
Page 4
Page 5
Page 6
Page 7
Page 8
Page 9
```
It is possible for me to do pre-processing before passing it off to Gooey, but it'd be nice to be able to have a total_process_regex supported as well, so I can grab the `Processing pages 1 through 1312` line.

**Comment by chriskiehl** [comment]
Please fill out the template and provide a code example. I can't really parse out what you're asking, what you've tried, or why it doesn't work. 

It looks like your output should be processable with the existing progress machinery, but, again, you haven't really provided enough details for me to judge one way or another : /

**Comment by Modelmat** [comment]
The current `progress_regex` requires each line to contain information about the total as well as the current. It is not possible to have one line have total and the next lines only state the current [ which is what my text screenshot was supposed to illustrate].

For instance, the https://github.com/chriskiehl/GooeyExamples/blob/b06b4742d7fa11db46fce29fde5ef1b697b0c17e/examples/example_progress_bar_3.py progress bar example here shows the regex as `r"^progress: (?P<current>\d+)/(?P<total>\d+)$"`. I do not see a way with this to have the regex parse and save the total for one line, then do the current for each other line.

Looking at the [code](https://github.com/chriskiehl/Gooey/blob/f87446ebc51b32050ffbfcb61e0a6aea657903ad/gooey/gui/processor.py#L78-L109), the system only allows for parsing the progress from values in the current line's regular expression. 

As a small example of what I cannot see how to do: [modified from the example_progress_bar_3]
```py
@Gooey(progress_regex=r"^(?P<current>\d+)/(?P<total>\d+)$",
       progress_expr="current / total * 100")
def main():
    parser = GooeyParser(prog="example_progress_bar_3")
    parser.add_argument("steps", type=int, default=15)
    parser.add_argument("delay", type=int, default=1)
    args = parser.parse_args(sys.argv[1:])

    print("Total: 100")

    for i in range(args.steps):
        print("Current: {}".format(i + 1, args.steps))
        sys.stdout.flush()
        sleep(args.delay)
```

For my code, it required me manually parsing the output and then forwarding it along to gooey: https://github.com/Modelmat/pdf-bookmarker/blob/a3ecade840027a4ccf5bfaacabb87738a8204c43/bookmarker.py#L170-L184

**Comment by chriskiehl** [comment]
Ah, ok. I understand now. Thanks for the additional details.  

So, your analysis is correct. Gooey only looks at the current line when applying the progress regex. This is basic, simple, somewhat restrictive, but satisfies most use cases. Additionally, as you mention, you can still use the current system by slightly modifying your output to include the total. 

This is probably about as "smart" as the progress code will get for the time being. 

Supporting your use case would be tricky. If you squint, to some degree, we'd be talking about building a tiny stateful expression interpreter. That `total` has to be (a) located in a stream, (b) bound to a variable in our 'progress interpreter' (that lives beyond the lifecycle of a single function call) and (c) made available to the environment where the expression is being evaluated. This is a lot more involved than simple `Line -> ProgressInfo` gizmo that happens now for each line. 

So, in short, that's gunna be a no from me, dogg. However, if you have a clean path towards adding this feature, a strong set of use cases, and a design overview for implementation, feel free to open a new issue and we could talk about it in depth!

**Comment by Modelmat** [comment]
Okay, thanks.
If it's not worth the effort to implement it in Gooey; would you be opposed to me creating an example that demonstrates how to do this (so other people would be able to see it?)
